app/vmui/packages/vmui/src/pages/ExploreAlerts/ExploreRules.tsx

```diff
@@ -0,0 +1,266 @@
+  );
+  const [searchInput, setSearchInput] = useState(defaultSearchInput);
+  const [types, setTypes] = useState(defaultTypes);
+  const [states, setStates] = useState(defaultStates);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const { hash } = useLocation();
+
+  if (!hash && groupId && (ruleId || alertId)) {
+    setModalOpen(true);
+  } else {
+    setModalOpen(false);
+  }
+
+  useSetQueryParams({
+    types: types.join("&"),
+    states: states.join("&"),
+    search: searchInput,
+    group_id: groupId,
+    alert_id: alertId,
+    rule_id: ruleId,
+  });
+
+  const handleChangeSearch = (input: string) => {
+    if (!input) {
+      setSearchInput("");
+    } else {
+      setSearchInput(input);
+    }
+  };
+
+  const handleGroupChangeExpand = useCallback(
+    (id: string) => (value: boolean) => {
+      setExpandGroups((prev) => {
+        const newExpandGroups = new Set(prev);
+        if (value) {
+          newExpandGroups.add(id);
+        } else {
+          newExpandGroups.delete(id);
+        }
+        return newExpandGroups;
+      });
+    },
+    [],
+  );
+
+  const handleRuleChangeExpand = useCallback(
+    (id: string) => (value: boolean) => {
+      setExpandRules((prev) => {
+        const newExpandRules = new Set(prev);
+        if (value) {
+          newExpandRules.add(id);
+        } else {
+          newExpandRules.delete(id);
+        }
+        return newExpandRules;
+      });
+    },
+    [],
+  );
+
+  const handleChangeStates = (title: string) => {
+    setStates(getChanges(title, states));
+  };
+
+  const handleChangeTypes = (title: string) => {
+    setTypes(getChanges(title, types));
+  };
+
+  let noRuleFound = "No rules found!";
+  if (typeFilter === "alert") {
+    noRuleFound = "No alerting rules found!";
+  } else if (typeFilter === "record") {
+    noRuleFound = "No recording rules found!";
+  }
+
+  const handleClose = () => {
+    searchParams.delete("group_id");
+    searchParams.delete("rule_id");
+    searchParams.delete("alert_id");
+    setSearchParams(searchParams);
+    setModalOpen(false);
+  };
+
+  if (modalOpen) {
+    const id = ruleId !== "" ? ruleId : alertId;
+    if (ruleId !== "") {
+      return (
+        <ExploreRule
+          groupId={groupId}
+          id={id}
+          mode={ruleId !== "" ? "rule" : "alert"}
+          onClose={handleClose}
+        />
+      );
+    } else {
+      return (
+        <ExploreAlert
+          groupId={groupId}
+          id={id}
+          mode={ruleId !== "" ? "rule" : "alert"}
+          onClose={handleClose}
+        />
+      );
+    }
+  } else {
+    const {
+      groups,
+      isLoading: loadingGroups,
+      error: errorGroups,
+    } = useFetchGroups({ typeFilter });
+
+    const isLoading = useMemo(() => {
+      return loadingGroups;
+    }, [loadingGroups]);
+
+    const error = useMemo(() => {
+      return errorGroups;
+    }, [errorGroups]);
+
```

Comment:
unnecessary `useMemo`
```suggestion
    const {
      groups,
      isLoading,
      error,
    } = useFetchGroups({ typeFilter });
```